ui/redux/selectors.js

```diff
@@ -65,94 +64,90 @@ export const getFamiliesGroupedByProjectGuid = createSelector(getFamiliesByGuid,
 export const getAnalysisGroupsGroupedByProjectGuid = createSelector(getAnalysisGroupsByGuid, groupEntitiesByProjectGuid)
 export const getSamplesGroupedByProjectGuid = createSelector(getSamplesByGuid, groupEntitiesByProjectGuid)
 
+const groupIndividualsByFamilyGuid = individuals =>
+  individuals.reduce((acc, individual) => {
+    if (!acc[individual.familyGuid]) {
+      acc[individual.familyGuid] = []
+    }
+    acc[individual.familyGuid].push(individual)
+    return acc
+  }, {})
+
 export const getIndividualsByFamily = createSelector(
   getIndividualsByGuid,
-  (individualsByGuid) => {
-    return Object.values(individualsByGuid).reduce((acc, individual) => {
-      if (!acc[individual.familyGuid]) {
-        acc[individual.familyGuid] = []
-      }
-      acc[individual.familyGuid].push(individual)
-      return acc
-    }, {})
-  },
+  individualsByGuid => groupIndividualsByFamilyGuid(Object.values(individualsByGuid)),
 )
 
-/**
- * function that returns a mapping of each familyGuid to an array of individuals in that family.
- * The array of individuals is in sorted order.
- *
- * @param state {object} global Redux state
- */
-export const getSortedIndividualsByFamily = createSelector(
-  getIndividualsByFamily,
+const getSortedIndividuals = createSelector(
+  getIndividualsByGuid,
   getMmeSubmissionsByGuid,
-  (familyIndividuals, mmeSubmissionsByGuid) => {
+  (individualsByGuid, mmeSubmissionsByGuid) => {
```

Comment:
sorting all the individuals, and then going through them in order and grouping them by family is WAY faster than grouping them and then sorting each grouped list. Made the same change for the sorted samples as well